src/TokenCache.ts

```diff
@@ -87,21 +79,14 @@ class TokenCache {
         };
 
         return getAccessToken(config)
-          .then((token: Token) => {
-
-            return getTokenInfo(this.oauthConfig.tokenInfoEndpoint, token.access_token);
-          })
-          .then((tokenInfo: TokenInfo) => {
+          .then((token) => getTokenInfo(this.oauthConfig.tokenInfoEndpoint, token.access_token))
+          .then((token) => {
 
-            tokenInfo.local_expiry = Date.now() + tokenInfo.expires_in * 1000 - EXPIRE_THRESHOLD;
-            this._tokens[tokenName] = tokenInfo;
-            resolve(tokenInfo);
+            token.local_expiry = Date.now() + token.expires_in * 1000 - EXPIRE_THRESHOLD;
+            this._tokens[tokenName] = token;
+            return Promise.resolve(token);
```

Comment:
Or just return token. Would leave a blank line to be aligned with our coding guidelines.